pkg/process/process_unix.go

```diff
@@ -0,0 +1,64 @@
+//go:build !windows
+// +build !windows
+
+package process
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+
+	"golang.org/x/sys/unix"
+)
+
+// Alive returns true if process with a given pid is running.
+func Alive(pid int) bool {
+	if pid < 1 {
+		return false
+	}
+	switch runtime.GOOS {
+	case "darwin":
+		// OS X does not have a proc filesystem.
+		// Use kill -0 pid to judge if the process exists.
+		err := unix.Kill(pid, 0)
+		if err == nil || err == unix.EPERM {
+			return true
+		}
+		return false
+	default:
+		if _, err := os.Stat(filepath.Join("/proc", strconv.Itoa(pid))); err == nil {
+			return true
+		}
+		return false
+	}
+}
+
+// Kill force-stops a process.
+func Kill(pid int) {
+	if pid < 1 {
+		return
+	}
+	_ = unix.Kill(pid, unix.SIGKILL)
```

Comment:
Since this is a new package without any back-compat concerns, thoughts on returning the error to the caller?

Reply:
I guess I did so to keep the old signature (to make it a drop-in replacement), but I can change it (and just make a thin wrapper for the alias) 👍

Reply:
Hmm.. actually getting back to this; wondering how useful the error would be; `SIGKILL` cannot be ignored, so the process "must" be terminated, so in which case would we get a useful error?

- no permissions (not running as `root` ?)
- process doesn't (no longer) exist; the intent of this function is to terminate the process, so this should be fine as well 🤔

Was there a specific error-case you had in mind?

Reply:
`EPERM` would absolutely be a useful error to return, because this function is inside a `pkg`-tree package. Since the policy is that `pkg`-tree packages are meant to be reusable outside of the daemon, functions in these packages shouldn't really be assuming they will be running as `root`. I think it would make sense to swallow `ESRCH` (no such process) because, as you point out, the intent of the call is to make the PID not exist so task failed successfully.

I think it would also be useful to not swallow any "impossible" errors such as `ENOSYS` or `EINVAL`. It's not outside the realm of possibility for some sysadmin to apply a seccomp filter or apparmor profile to the daemon which blocks the kill syscall, for example. Though the calling process would be running in a very insane environment if SIGKILL is an invalid signal or the syscall is returning nonsensical errors; perhaps the only sane thing to do in an insane world is to `panic()`?